searxng/searxng
# [fix] make /stats more CSP compliant

## What does this PR do?

Make /stats more CSP compliant, for details read commit messages.

## Why is this change important?

Reported-by: #57

## How to test this PR locally?

Test locally (see below), before we merge I will test this PR with CSP _activated_ on my [public instance](https://darmarit.org/searx/stats).

Aktivate checker ..

```diff
modified   searx/settings.yml
@@ -106,7 +106,7 @@ outgoing: # communication with search engines
 
 checker:
     # disable checker when in debug mode
-    off_when_debug: True
+    off_when_debug: False
 
     # scheduling: interval or int
     # use "scheduling: False" to disable scheduling
@@ -114,9 +114,9 @@ checker:
     # * uncomment "scheduling" section
     # * add "cache2 = name=searxcache,items=2000,blocks=2000,blocksize=4096,bitmap=1" to your uwsgi.ini
 
-    # scheduling:
-    #    start_after: [300, 1800]  # delay to start the first run of the checker
-    #    every: [86400, 90000]  # how often the checker runs
+    scheduling:
+        start_after: [10, 20]  # delay to start the first run of the checker
+        every: [1000, 2000]  # how often the checker runs
 
     # additional tests: only for the YAML anchors (see the engines section)
```

Start instance ..

    make run 

Doe some queries, wait until checker has finished .. and after visit /stats page
